### PR TITLE
fix(inline-notification): add 8px right padding if close button is hidden

### LINKED ISSUE #4864: [InlineNotification]: Action button is misaligned when close button is hidden
- [x] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

When I pass `hideCloseButton` prop to `InlineNotification` and have an action button set up, the button is too far to the right so that its background is flush to the right side when hovering.

## Steps to reproduce the issue

1. Visit storybook at http://react.carbondesignsystem.com/?path=/story/notifications--inline
2. Select knob `hideCloseButton`
3. Hover over "Action" buttton
(better visible when `lowContrast` is also selected)

## Additional information

image
image

### PR DESCRIPTION
**Pull request by jendowns**
Closes #4864 

Right now if you remove the close button in the `InlineNotification` (as shown in screenshots for #4864), the action button will be flush against the right side of the notification. For the low-contrast variation, this flush right margin is problematic because it overlaps the 1px low-contrast border.

~This PR proposes adding a 1px right margin to the action button so that it won't be "flush" on the right side of the notification & also won't overlap the low-contrast border:~ Updated to 8px now

#### Changelog

**Changed**

- add 8px right padding to the `InlineNotification` if `hideCloseButton={true}`

### CODE REVIEW

**Review by joshblack — APPROVED** [review]

**Review by laurenmrice — CHANGES_REQUESTED** [review]
If there is no close icon can we adjust the action button to be 8px away from the right hand side.

Screen Shot 2019-12-13 at 1 43 53 PM

**Comment by jendowns** [comment]
Thank you @laurenmrice! I just committed an update to the right spacing so that should be reflected in the previews soonl. 👍 

@joshblack I ended up changing how I added spacing... since the 8px spacing would only be present if `hidenCloseButton={true}`, I moved that logic to the component js. Could you take another look at what I did when you get a chance & let me know if you agree? Thank you!

File: packages/react/src/components/Notification/Notification.js [hunk 1]
```diff
@@ -504,7 +505,12 @@ export class InlineNotification extends Component {
     );
 
     return (
-      <div {...other} role={role} kind={kind} className={classes}>
+      <div
+        {...other}
+        role={role}
+        kind={kind}
+        className={classes}
+        style={{ paddingRight: hideCloseButton ? spacing03 : '0' }}>
```
Comment (asudoh) [inline packages/react/src/components/Notification/Notification.js]: Any reason you didn't do this via Sass code (and update `className` upon `hideCloseButton`)...?

**Review by asudoh — COMMENTED** [review]
Thank you very much @jendowns for fixing!

File: packages/react/src/components/Notification/Notification.js (hunk 1 above)
Comment (jendowns) [reply packages/react/src/components/Notification/Notification.js]: Thanks @asudoh -- with the current structure of this component, the action button is a sibling to the close button & therefore it's difficult to target & change the action button when the close button is no longer present.

Adding a class is definitely a good alternative approach. I just went with the solution that (seemed to) add/remove the least. Would you prefer to add a class and keep the padding update in Sass only?

Comment (asudoh) [reply packages/react/src/components/Notification/Notification.js]: Thank you for sharing your thoughts @jendowns! I'm thinking about the ability of style override by applications, which tends to lean toward Sass approach. Is it changing `className` upon `hideCloseButton` enough to tell the Sass code that the close button is no longer present? If my words here is not clear enough please don't hesitate to speak up - Thanks!

Comment (jendowns) [reply packages/react/src/components/Notification/Notification.js]: That's a good point @asudoh -- I just committed a change that goes the classname + Sass route. Let me know what you think -- thank you!

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @jendowns for the update!

**Review by laurenmrice — CHANGES_REQUESTED** [review]
Action link placement looks good. Can we take off the Example Link that comes after the normal text? I don't know if that happened in this pr, but I would rather only show the action link on the right.

**Comment by jendowns** [comment]
Thanks @laurenmrice! If you don't mind, I was wondering if we could keep those example links there? 😅 

I added them in a PR a week or so ago because a change went through that prevented clicking inside the notification body text. Without the example link there previously, the change went through undetected & had an effect on both `gatsby-theme-carbon` (mentioned here: https://github.com/carbon-design-system/carbon/issues/4804) and the `carbon-website` (mentioned here: https://github.com/carbon-design-system/carbon-website/issues/655). 

OH and I should add: technically since that notification body, or specifically the `subtitle`, accepts a `node` type, it can have a link passed into it by a user. Which is what we were seeing on both the Gatsby theme and the Carbon official website.

So keeping that link there could help catch any future issues with links in the body. 😅 What do you think?

**Review by joshblack — APPROVED** [review]

**Review by laurenmrice — APPROVED** [review]
Okay we can keep it there thanks for explaining, will approve. I am wondering if it could eventually be something that you could turn off and on from the story? I am just wondering if people will be confused why there is a link and also an action link.

**Review by dakahn — APPROVED** [review]

**Comment by jendowns** [comment]
@laurenmrice Yeah that's a good question... I'm honestly not sure if storybook knobs can allow you to customize HTML elements (specifically the `<a>` element for this example link) which may make it hard to customize that `InlineNotification` body/`subtitle` nodes 🤔 

BUT another option: I could add a storybook variation? Something you select on the sidebar? Right now there's `Toast` and `Inline`... there could be a new one like `Inline without example link` or the inverse, `Inline with example link` -- too much?

**Comment by laurenmrice** [comment]
I think we can leave it for now and I can revisit that eventually :) thanks jen!

**Pull request merged**